matchpy/matching/bipartite.py

```diff
@@ -164,7 +164,7 @@ def find_matching(self) -> Dict[TLeft, TRight]:
             else:
                 directed_graph[tail].add(head)
 
-        matching = HopcroftKarp(directed_graph).maximum_matching()
+        matching = HopcroftKarp(directed_graph).get_maximum_matching()
```

Comment:
`directed_graph` is defined as being of type `directed_graph = {}  # type: Dict[Tuple[int, TLeft], Set[Tuple[int, TRight]]]` on line 157. In the class `HopcroftKarp` there is therefore a redefinition of types:
- `Tuple[int, TLeft] ==> TLeft`
- `Tuple[int, TRight] ==> TRight`
maybe we should rename `TLeft` and `TRight`.

Anyway, it's MatchPy that's representing graphs as dictionaries mapping to sets, so maybe the input should be kept this way.

Reply:
Yes, I guess it makes sense to rename the types.

> Anyway, it's MatchPy that's representing graphs as dictionaries mapping to sets, so maybe the input should be kept this way.

Since this representation is only constructed to be used as input to the Hopcroft-Karp algorithm, I don't mind changing the representation to a dictionary mapping to lists.

Reply:
> Since this representation is only constructed to be used as input to the Hopcroft-Karp algorithm, I don't mind changing the representation to a dictionary mapping to lists.

OK. That's nice to hear.